CRM/Contribute/BAO/Contribution.php

```diff
@@ -2639,7 +2639,7 @@ public function _gatherMessageValues($input, &$values, $ids = array()) {
       }
       // set lineItem for contribution
       if ($this->id) {
-        $lineItem = CRM_Price_BAO_LineItem::getLineItems($this->id, 'contribution', 1);
+        $lineItem = CRM_Price_BAO_LineItem::getLineItemsByContributionID($this->id, TRUE);
```

Comment:
I tested removing the 'TRUE' on this line locally and then generating message receipts through the 'print contribution receipts' action off advanced search. My conclusion - it is currently broken & not displaying membership line items.

This commit https://github.com/civicrm/civicrm-core/pull/9327 is my fix for it - using your principle but moving the quick config concept up to the message template and letting the template decide whether to show the line items.